### ISSUE TITLE
Cursor property doesn't work in React Native macOS

### ISSUE THREAD

**Issue by psalishol** [microsoft/react-native-macos#1804]
### Environment

```markdown
react-native -v:
npm ls react-native-macos:
node -v:
npm -v:
yarn --version:
xcodebuild -version:
```

### Steps to reproduce the bug

The cursor property in React Native macOS does not work as expected. When using the cursor property to change the appearance of the cursor on certain UI elements, such as buttons or links, the cursor type remains the default arrow cursor and does not change to the expected cursor type.

This behavior is inconsistent with the way the cursor property works on web platforms, where it changes the cursor type as expected. This inconsistency can cause confusion for users, who may expect the cursor to behave the same way across different platforms.

Steps to Reproduce:

1. Create a new React Native macOS project
2. Add a UI element such as a button or link
3. Set the cursor property to a non-default value, such as "pointer"
4. Run the project and hover over the UI element

### Expected Behavior

The cursor type changes to the specified value when hovering over the UI element.

### Actual Behavior

The cursor type remains the default arrow cursor

**Comment by Saadnajmi** [comment]
You're setting the cursor property of the style prop to one of these values? 
https://github.com/microsoft/react-native-macos/blob/main/React/Views/RCTCursor.m#L15-L34

**Comment by psalishol** [comment]
i used one of the cursor properties, but the issue still remain

https://user-images.githubusercontent.com/85138073/235318213-1210f731-3880-46fb-a6a7-c85b12f2b48c.mov

Here is the component that renders that:

```
const TestCursorProperty = () => {
    return (
        <View
             style={{
             height: 100,
             width: "100%",
             backgroundColor: "red",
             cursor: "e-resize",
               }}
            >
             <Text>This is for testing</Text>
        </View>
     )};
```

**Comment by Saadnajmi** [comment]
Are you on 0.71?

**Comment by psalishol** [comment]
> Are you on 0.71?

oh i forgot to indicate the react native version.

i'm on 0.64

**Comment by Saadnajmi** [comment]
> > Are you on 0.71?
> 
> oh i forgot to indicate the react native version.
> 
> i'm on 0.64

Ah... I'm pretty sure we only have it implemented for 0.71+

**Comment by psalishol** [comment]
> > > Are you on 0.71?
> > 
> > 
> > oh i forgot to indicate the react native version.
> > i'm on 0.64
> 
> Ah... I'm pretty sure we only have it implemented for 0.71+

It works for 0.71+

**Comment by taenri** [comment]
I am still seeing this issue on v0.71.6, but only when the View wrapping the text element perfectly fits the text.

It seems the cursor property cannot be set directly for Text elements. Since children are displayed on top of their parents, this means perfectly wrapped Text can never update the cursor properly.

```
const TestCursorProperty = () => {
    return (
      <View style={{flexGrow: 0, flexShrink: 1, backgroundColor: 'red'}} cursor='pointer'>
        <Text>
          This is text for testing.
        </Text>
      </View>
     )};
```
